ft: add docs clarifying where to run prefect-deploy and entrypoint format (#11624)
PrefectHQ/prefect: docs/guides/prefect-deploy.md

    Any deployment configuration can be overridden via options available on the `prefect deploy` CLI command when creating a deployment.

>     !!! tip "New `prefect.yaml` file flexibility"

Reviewer: ```suggestion
    !!! tip "`prefect.yaml` file flexibility"
```

>         Traditionally, this file had to be in the root of your repository or project directory and named `prefect.yaml`.

Reviewer: ```suggestion
        In older versions of Prefect, this file had to be in the root of your repository or project directory and named `prefect.yaml`.
```

>         Now this file can be located elsewhere and be named differently. To use a custom prefect.yaml file, run the new 

Reviewer: ```suggestion
        Now this file can be located elsewhere and be uniquely named. To use a custom `prefect.yaml` file, run the new 
```

Author: My interpretation of 'uniquely named' is that the file has to be unique among, but it's technically possible for two `prefect.yaml` files to have the same name but be in different directories. Do you think the 'uniquely named' phrasing is still the best way to describe this?

Reviewer: ah, good call. I'll remove my suggestion.

Reviewer 2: wow I actually didn't realize that @urimandujano. that's actually huge. is that something to advertise?

Author: Let me add a note here to address that explicitly. If that was unclear to us then it'll definitely be unclear to others.

Reviewer 2: ty!

>     !!! tip "New `prefect.yaml` file flexibility"
>         Traditionally, this file had to be in the root of your repository or project directory and named `prefect.yaml`.
>         Now this file can be located elsewhere and be named differently. To use a custom prefect.yaml file, run the new 
>         form of the deploy CLI command from the root of your project directory: 
>         
>         `prefect deploy --prefect-file path/to/my_file.yaml`
> 

Reviewer 2: ```suggestion
    !!! tip "New `prefect.yaml` file flexibility"
        Traditionally, this file had to be in the root of your repository or project directory and named `prefect.yaml`.
        Now this file can be located elsewhere and named differently. To use a custom deployment configuration file, run the new 
        form of the deploy CLI command from the root of your project directory: 
        
        `prefect deploy --prefect-file path/to/my_file.yaml`

```

is saying "run the new form of the deploy CLI command" necessary? it seems like you'd just pass in the `--prefect-file` option?

Author: I think that "new form" is a grandiose way of saying "just pass in a new flag" :) Let me update that to be a little less over the top!

    Any deployment configuration can be overridden via options available on the `prefect deploy` CLI command when creating a deployment.

    !!! tip "`prefect.yaml` file flexibility"

>         In older versions of Prefect, this file had to be in the root of your repository or project directory and named `prefect.yaml`. Now this file can be located in a directory outside the project or a subdirectory inside the project and be named differently. You can even have multiple `prefect.yaml` files with the same name in different directories. By default, `prefect deploy` will use a `prefect.yaml` file in the project's root directory. To use a custom  `prefect.yaml` file, supply the new  `--prefect-file` CLI argument when running the `deploy` command from the root of your project directory: 

Reviewer 2: ```suggestion
        In older versions of Prefect, this file had to be in the root of your repository or project directory and named `prefect.yaml`. Now this file can be located in a directory outside the project or a subdirectory inside the project. It can be named differently, provided the filename ends in `.yaml`. You can even have multiple `prefect.yaml` files with the same name in different directories. By default, `prefect deploy` will use a `prefect.yaml` file in the project's root directory. To use a custom deployment configuration file, supply the new  `--prefect-file` CLI argument when running the `deploy` command from the root of your project directory: 
```

    !!! tip "`prefect.yaml` file flexibility"
        In older versions of Prefect, this file had to be in the root of your repository or project directory and named `prefect.yaml`.

>         Now this file can be located elsewhere and be named differently. You can even have multiple `prefect.yaml` files with the same name in different directories. To use a custom `prefect.yaml` file, supply the new 

Reviewer 2: it by default deploys from root dir one named `prefect.yaml` right?

Author: Yup, if there's no flag the it defaults to `prefect.yaml` in the dir where `prefect deploy` is getting run (which should be the project's root dir). Are you thinking that's something I should mention?

Reviewer 2: i think so, yeah

Author: I mentioned that too!

Reviewer 2: ty!

>         Now this file can be located elsewhere and be named differently. You can even have multiple `prefect.yaml` files with the same name in different directories. To use a custom `prefect.yaml` file, supply the new 

Reviewer 2: by "elsewhere" it still needs to be in the root dir right? just don't want anyone thinking you could go from like a link to another repo

Author: It can be outside of the root dir (i.e in a parent dir  `../prefect.yaml` or subdir `./dev/prefect.yaml`. But it does need to be in a directory. What do you think of this change:

```diff
-- Now this file can be located elsewhere and be named differently.
++ Now this file can be located in another directory and be named differently.
```

Reviewer 2: i didn't know it could be outside the root dir, can we just say that explictly? 👼

Author: Done!

Reviewer 2: ty!

    Prefect supports multiple deployment declarations within the `prefect.yaml` file. This method of declaring multiple deployments allows the configuration for all deployments to be version controlled and deployed with a single command.

    New deployment declarations can be added to the `prefect.yaml` file by adding a new entry to the `deployments` list. Each deployment declaration must have a unique `name` field which is used to select deployment declarations when using the `prefect deploy` command.
    
    !!! warning

>         When using a custom `prefect.yaml` file, remember that the value for deployment `entrypoints` must be relative to 

Reviewer 2: can we clarify what custom means here for the user? a diff name or location?

Author: Both! I added a clarification.

Reviewer 2: ty!

>         When using a custom `prefect.yaml` file, remember that the value for deployment `entrypoints` must be relative to 

Reviewer 2: ```suggestion
        When using a custom `prefect.yaml` file, remember that the value for the deployment `entrypoint` must be relative to 
```

Reviewer 2: since I don't believe there is such a thing as `entrypoints` key (to my knowledge!), didn't want to confuse the user
